app/controllers/reviews_controller.rb

```diff
@@ -1,19 +1,28 @@
 class ReviewsController < ApplicationController
+  before_action :authorize_non_member, only: [:update, :edit, :destroy]
+
   def new
     @piece = Piece.find(params[:piece_id])
     @review = Review.new
   end
 
   def create
-    @piece = Piece.find(params[:piece_id])
-    @review = @piece.reviews.new(review_params)
-    @review.user = current_user
-    if @review.save
-      flash[:notice] = 'Your review has been successfully added.'
-      redirect_to piece_path(@piece)
-    else
-      flash[:notice] = @review.errors.messages.values.join(". ")
+    if !user_signed_in?
+      flash.now[:notice] = "Please sign in to review a piece of music"
+      @piece = Piece.find(params[:piece_id])
+      @review = Review.new
       render :'pieces/show'
+    else
```

Comment:
Revert this action to how it was, if a use is not authenticated then saving the review will fail on the user validation of the review, and hence they'll see the appropriate error message.